src/Scan/RestScans.ts

```diff
@@ -117,15 +117,21 @@ export class RestScans implements Scans {
     };
   }
 
-  private exploreDiscovery(body: ScanConfig): Discovery[] {
+  private async exploreDiscovery(body: ScanConfig): Promise<Discovery[]> {
     const discoveryTypes: Discovery[] = [];
 
     if (Array.isArray(body.crawlerUrls)) {
       discoveryTypes.push(Discovery.CRAWLER);
     }
 
     if (body.fileId) {
-      discoveryTypes.push(Discovery.ARCHIVE);
+      const file: StorageFile = await this.client.get({
+        uri: `/api/v1/files/${body.fileId}`
+      });
+
+      file.type !== 'har'
```

Comment:
Use if/else statement instead of the ternary operator. You can use it only when you are interested in the result of execution:

```ts
discoveryTypes.push(
  file.type === 'har' ? Discovery.ARCHIVE : Discovery.OAS
)
```